src/internal/cluster/tunnel.go

```diff
@@ -167,16 +167,16 @@ func ServiceInfoFromServiceURL(serviceURL string) *ServiceInfo {
 
 	// Match hostname against local cluster service format.
 	pattern := regexp.MustCompile(serviceURLPattern)
-	matches := pattern.FindStringSubmatch(parsedURL.Hostname())
+	get, err := utils.MatchRegex(pattern, parsedURL.Hostname())
 
 	// If incomplete match, return an error.
-	if len(matches) != 3 {
+	if err != nil {
 		return nil
```

Comment:
This seems odd... if there's an error, don't return an error?

Reply:
In the places we use this we don't really use the error but I refactored it to include errors anyway to hopefully help explain what parts of the functions are doing in case they get used for other things in the future.